brickstrap-image.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+#
+# This file is part of brickstrap.
+#
+# brickstrap - create a foreign architecture rootfs using kernel namespaces,
+#              multistrap, and qemu usermode emulation and create a disk image
+#              using libguestfs
+#
+# Copyright (C) 2016 Johan Ouwerkerk <jm.ouwerkerk@gmail.com>
+#
+
+#
+# Note: this file is not meant to be executed, source it as a library of
+# functions instead. Variables used by the functions (other than stack) are
+# namespaced using the 'BR_' or 'BRP_' prefixes. Function names are namespaced
+# similarly, using the 'br_' and 'brp_' prefixes. See docs/namespacing.md for
+# more information on namespacing in brickstrap.
+#
+
+#
+# Default image type 'driver'. This function simply delegates to the current
+# brickstrap default.
+# $1: path to the image file to generate.
+#
+function brp_image_default_driver()
+{
+    brp_image_drv_bootroot "$1"
+}
+
+#
+# Default image type configuration validator. This function simply delegates to
+# the current brickstrap default.
+#
+function brp_validate_default_image_config()
+{
+    true # there is no validation required for the simple bootroot driver, yet
+}
+
+#
+# Displays an error message if a image type driver failed to create an image
+# file successfully. Returns the given status code.
+# $1: the status code returned by the driver function
+# $2: the name of the partitioning/imaging scheme implemented in the disk image
+# $3: the file path of the disk image
+#
+function brp_image_err_msg()
+{
+    error "Unable to create image: '$3'.
+Driver '$2' ($(brp_get_image_type_driver "$2")) returned status code: $1"
+    return $1
+}
+
+#
+# Create a disk image file (type). This function looks up the
+# relevant driver and invokes it with the given path. Before creating the image
+# this function checks it does not already exist or, if it does, that it may be
+# overwritten.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+#
+function brp_create_image_type()
+{
+    if [ $# -eq 0 -o -z "$1" ]; then
+        error "Image type required!"
+        return 1
+    elif [ -z "$(brp_get_image_type_driver "$1")" ]; then
+        error "Unable to create image: '$2'.
+No driver for image type: '$1'"
+        return 1
+    fi
+    BRP_CUR_IMG="$(brp_image_path "$1" "$(brp_get_image_type_extension $1)")"
+    debug "IMAGE: $BRP_CUR_IMG"
+    if [ -z "$BR_FORCE" -a -f "$BRP_CUR_IMG" ]; then
+        error "$BRP_CUR_IMG already exists. Use -f option to overwrite."
+        return 1
+    else
+        eval "$(brp_get_image_type_driver "$1")" "$BRP_CUR_IMG" || \
+            brp_image_err_msg "$?" "$1" "$BRP_CUR_IMG"
+    fi
+}
+
+#
+# Look up the driver function for a given image type name.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+#
+function brp_get_image_type_driver()
+{
+    [ $# -eq 1 -a -n "$1" ] && case "$1" in
+        default)
+            echo -n brp_image_default_driver
```

Comment:
Are you missing `;;` here? I just tried running brickstrap from this branch and got

    brickstrap-image.sh: line 90: syntax error near unexpected token `)'

Works after I add `;;`